### PR TITLE
Add origin to Next Queries Preview

### PR DESCRIPTION
**Pull request by annacv**
[EX-6757](https://searchbroker.atlassian.net/browse/EX-6757)

# Pull request template
 
- Modify `fetchAndSaveNextQueryPreviewWire` to send both the `query` and the `location` in the request.
- Update store `fetchAndSaveNextQueryPreview`and `fetchNextQueryPreview` actions & its types.
- Update tests.
- Create a test for an undefined location use case

## Motivation and context

Enable sending an `origin` for this feature to be tracked.

- [ ] Dependencies. If any, specify:
- [ ] Open issue. If applicable, link:

## Type of change

- [ ] Bug fix (non-breaking change that fixes an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that causes existing functionality to not work as expected)
- [ ] Change requires a documentation update

## What is the destination branch of this PR?

- [x] `Main`
- [ ] Other. Specify:

## How has this been tested?

Tests performed according to [testing guidelines](./contributing/tests.md): 

## Checklist:

- [x] My code follows the **[style guidelines](./CONTRIBUTING.md#style-guides)** of this project.
- [x] I have performed a **self-review** on my own code.
- [x] I have **commented** my code, particularly in hard-to-understand areas.
- [ ] I have made corresponding changes to the **[documentation](./CONTRIBUTING.md#documentation-style-guide)**.
- [x] My changes generate **no new warnings**.
- [x] I have added **tests** that prove my fix is effective or that my feature works.
- [x] New and existing **unit tests pass locally** with my changes.

### CODE REVIEW

File: packages/x-components/src/x-modules/next-queries/store/__tests__/actions.spec.ts [hunk 1]
```diff
@@ -35,7 +36,7 @@ describe('testing next queries module actions', () => {
     NextQueriesGetters,
     NextQueriesMutations,
     NextQueriesActions
-  > = new Store(nextQueriesXStoreModule as any);
+  > = new Store(nextQueriesXStoreModule as XStoreModule<any, any, any, any>);
```
Comment (diegopf) [inline packages/x-components/src/x-modules/next-queries/store/__tests__/actions.spec.ts:39]: You can undo those changes, they are not needed.

File: packages/x-components/src/x-modules/next-queries/store/actions/fetch-and-save-next-query-preview.action.ts [hunk 2]
```diff
@@ -2,13 +2,13 @@ import { NextQueriesXStoreModule } from '../types';
...
-  ({ dispatch, commit }, query) => {
-    return dispatch('fetchNextQueryPreview', query)
+  ({ dispatch, commit }, { query, location }) => {
+    return dispatch('fetchNextQueryPreview', { query, location })
       .then(response => {
         if (response) {
           commit('setResultsPreview', {
             [query]: {
-              query,
+              query: query,
```
Comment (diegopf) [inline packages/x-components/src/x-modules/next-queries/store/actions/fetch-and-save-next-query-preview.action.ts:11]:
```suggestion
              query,
```

File: packages/x-components/src/x-modules/next-queries/store/actions/fetch-next-query-preview.action.ts [hunk 3]
```diff
@@ -1,27 +1,32 @@
...
  */
 export const fetchNextQueryPreview: NextQueriesXStoreModule['actions']['fetchNextQueryPreview'] = (
   { state },
-  query
+  { query, location }
 ) => {
   if (!query) {
     return null;
   }
+  const origin = createOrigin({ feature: 'next_query', location: location });
```
Comment (diegopf) [inline packages/x-components/src/x-modules/next-queries/store/actions/fetch-next-query-preview.action.ts:22]:
```suggestion
  const origin = createOrigin({ feature: 'next_query', location: location }) ?? undefined;
```

File: packages/x-components/src/x-modules/next-queries/store/actions/fetch-next-query-preview.action.ts [hunk 4]
```diff
@@ -1,27 +1,32 @@
...
+
   return XPlugin.adapter.search(
     {
       query,
       rows: state.config.maxPreviewItemsToRequest,
-      extraParams: state.params
+      extraParams: state.params,
+      ...(origin && { origin })
```
Comment (diegopf) [inline packages/x-components/src/x-modules/next-queries/store/actions/fetch-next-query-preview.action.ts:29]:
```suggestion
      origin
```

**Review by diegopf — CHANGES_REQUESTED** [review]

File: packages/x-components/src/x-modules/next-queries/store/__tests__/actions.spec.ts (hunk 1 above)
Comment (annacv) [reply packages/x-components/src/x-modules/next-queries/store/__tests__/actions.spec.ts:39]: Sure? ESLint warned `Unsafe argument of type 'any' assigned to a parameter`

File: packages/x-components/src/x-modules/next-queries/store/actions/fetch-next-query-preview.action.ts [hunk 5]
```diff
@@ -1,27 +1,32 @@
...
  */
 export const fetchNextQueryPreview: NextQueriesXStoreModule['actions']['fetchNextQueryPreview'] = (
   { state },
-  query
+  { query, location }
 ) => {
   if (!query) {
     return null;
   }
+  const origin = createOrigin({ feature: 'next_query', location: location }) ?? undefined;
```
Comment (LuisMartinez15) [inline packages/x-components/src/x-modules/next-queries/store/actions/fetch-next-query-preview.action.ts:22]:
```suggestion
  const origin = createOrigin({ feature: 'next_query', location }) ?? undefined;
```

File: packages/x-components/src/x-modules/next-queries/store/actions/fetch-next-query-preview.action.ts [hunk 6]
```diff
@@ -1,27 +1,32 @@
 import { NextQueriesXStoreModule } from '../types';
 import { XPlugin } from '../../../../plugins/x-plugin';
+import { createOrigin } from '../../../../utils/index';
```
Comment (LuisMartinez15) [inline packages/x-components/src/x-modules/next-queries/store/actions/fetch-next-query-preview.action.ts:3]: Avoid importing from barrel

```suggestion
import { createOrigin } from '../../../../utils/origin';
```

**Comment by diegopf** [comment]
could you please add the location provider to the `SlidingQueryPreview ` in the [home view](https://github.com/empathyco/x/blob/feature/EX-6757_Add-origin-NQ-preview/packages/x-components/src/views/home/Home.vue#L219)? Currently is missing and if you open the home you will get some error in the console

**Review by diegopf — APPROVED** [review]

**Review by LuisMartinez15 — COMMENTED** [review]
good job!

**Review by LuisMartinez15 — APPROVED** [review]

**Pull request merged**